### ISSUE TITLE
Dragonfly + operator make several incorrect assumptions about TLS configuration

### ISSUE THREAD

**Issue by solidDoWant** [dragonflydb/dragonfly#4171]
**Describe the bug**

There are several bugs related to Dragonfly TLS configuration when used in conjunction with the Dragonfly operator:
1. [Here](https://github.com/dragonflydb/dragonfly/blob/main/tools/docker/healthcheck.sh#L24) and [here](https://github.com/dragonflydb/dragonfly/blob/main/tools/docker/healthcheck.sh#L26) Dragonfly assumes that if a file exists at `/etc/dragonfly/tls/ca.crt` then TLS is enabled on the health check port. However, the operator the operator uses this directory [as defined here](https://github.com/dragonflydb/dragonfly-operator/blob/main/internal/resources/resources.go#L39) for the client CA cert. If the serving cert and the client cert don't have a CA in common (and they really shouldn't in many cases), then the health check will fail.
2. The operator [disables TLS on the admin port here](https://github.com/dragonflydb/dragonfly-operator/blob/main/internal/resources/resources.go#L262) when server auth TLS is enabled, but (as per the above) the Dragonfly health check expects it to be enabled.
3. The operator [specifies the path to the server auth TLS certificate here](https://github.com/dragonflydb/dragonfly-operator/blob/main/internal/resources/resources.go#L264) as `tls.crt`, but per (1) the health check assumes that it is listed at `ca.crt`.

**To Reproduce**
1. Deploy the dragonfly operator
2. Deploy an instance with [this config](https://github.com/solidDoWant/infra-mk3/blob/c57ce000398270256b2e94c449c5570627eb45a3/cluster/gitops/security/authentik/backend/redis/cluster.yaml) and [these certificates](https://github.com/solidDoWant/infra-mk3/tree/master/cluster/gitops/security/authentik/backend/redis)
3. Watch the health check fail repeatedly due to the above

**Expected behavior**

Dragonfly should not crash when configured for TLS auth

**Screenshots**

**Environment (please complete the following information):**
 - OS: [ubuntu 20.04] Talos
 - Kernel: # Command: `uname -a` 6.6
 - Containerized?: [Bare Metal, Docker, Docker Compose, Docker Swarm, Kubernetes, Other] k8s
 - Dragonfly Version: [e.g. 0.3.0] v1.24.0, operator v0.13.1

**Reproducible Code Snippet**
```
See reproduction steps
```

**Additional context**

**Comment by romange** [comment]
@Pothulapati  PTAL.


@solidDoWant  you wrote "Dragonfly should not crash when configured for TLS auth". Does it indeed crash?

**Comment by solidDoWant** [comment]
@romange Sorry, I phrased this poorly. The dragonfly process itself does not crash. However, the container's health check fails, which causes the container orchestrator (k8s) to terminate the pod/container.

**Comment by romange** [comment]
@solidDoWant I understand that the code in healthcheck is far from perfect but it seems that with some workarounds it is possible to make it work. For example, per (2) the admin port does use TLS. You can just pas the environment variable `HEALTHCHECK_PORT=<admin_port>` to your docker container in order to instruct the health checker to probe the admin port using `nc`.

**Comment by solidDoWant** [comment]
There's a few of problems with this:
* Per (2) the admin port does _not_ use TLS (ever with the operator), as the linked line shows that the `--no_tls_on_admin_port` flag is set
* [The healthcheck port is already set by the operator to the admin port](https://github.com/dragonflydb/dragonfly-operator/blob/main/internal/resources/resources.go#L117-L120)
* I cannot change the healthcheck port even if I wanted to, [because the operator does not expose this as a configurable value](https://github.com/dragonflydb/dragonfly-operator/blob/main/internal/resources/const.go#L27)
* Even if TLS was enabled on the port, the health check still uses the wrong CA (it uses the client auth cert, not the serving cert) for the health check, causing the `openssl` command to fail with a cert chain validation error

I'm open to workarounds, but this doesn't solve the problems.

**Comment by romange** [comment]
I recently introduced a usability improvement to Dragonfly that allows it to respond to plain requests on the tls port with `-ERR Bad TLS header, double check if you enabled TLS for your client.`

therefore a simple nc should work well for both tls and non-tls protocols now. 
I just verified,  `echo PING | nc localhost 6379` returns with 0 fort tls port as well. 
I will change the health check to use only nc - this should solve the problem and will make the everything more simple.
